src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -784,6 +787,13 @@ public boolean advanceExact(int doc) throws IOException {
         doubleValues.advanceExact(docID);
         compositeFieldValue.addFieldValue(
             FieldValue.newBuilder().setDoubleValue(doubleValues.doubleValue()));
+      } else if (fd instanceof RuntimeFieldDef) {
+        RuntimeFieldDef runtimeFieldDef = (RuntimeFieldDef) fd;
+
+        assert !Double.isNaN(hit.getScore()) || runtimeFieldDef.getValuesSource() != null;
+
+        Object obj = runtimeFieldDef.getValuesSource();
+        compositeFieldValue.addFieldValue(FieldValue.newBuilder().setStructValue((Struct) obj));
```

Comment:
Does this work? The object type would be `ValueSource`, but it is cast to a `Struct`?